config/v1/types_infrastructure.go

```diff
@@ -478,6 +478,11 @@ type AWSPlatformStatus struct {
 	// +kubebuilder:validation:MaxItems=25
 	// +optional
 	ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+
+	// userDNS contains user provided configuration for API and API-Int
+	// Load Balancers when user configured DNS is used in place of Route53.
+	// +optional
+	UserDNS UserConfiguredDNS `json:"userDNS,omitempty"`
```

Comment:
I think we want to change this a little, so it's less specific, I think the end result should look like this in a normal cluster
```
dns:
  provider: Default
```
or like below for the user configured version.
```
dns:
  provider: User
  user:
    apiLoadBalancer:
      name: foo
    apiInternalLoadBalancer:
       name: bar
```
This would implement a discriminated union for the configuration, example [here](https://github.com/openshift/api/blob/391aead809e1374e9978feeb1305a3479523aadc/example/v1/types_stable.go#L88-L105)

This gives us the flexibility in the future to expand configuration for the default DNS provider should we need to

Reply:
The additional layer of `user` is required for it to be a [discriminated union](https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#discriminated-unions), KEP for reference: https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1027-api-unions/README.md,

Reply:
Just making sure we want the discriminated union even when the additions to the API are in the "Status" and not the "Spec". (Double checking because all examples are for the Spec.)

Reply:
I think we want it in the status too, makes it clear. Spec and status are equally editable by end users these days, even though that's not that desirable